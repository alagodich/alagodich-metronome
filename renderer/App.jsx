import MainWindow from './components/MainWindow.jsx';
import React from 'react';
import ReactDom from 'react-dom';

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    ReactDom.render(<MainWindow />, container);
};
