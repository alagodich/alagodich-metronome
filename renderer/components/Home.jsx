import React, {Component} from 'react';
import {Link} from 'react-router';

class Home extends Component {
    render() {
        return (
            <div className="ui basic segment">
                <div className="ui text container">
                    <h1 className="ui inverted header">{'Home'}</h1>
                    <div className="ui vertical text menu">
                        <Link to="/counter" className="item">{'Counter'}</Link>
                        <Link to="/metronome" className="item">{'Metronome'}</Link>
                        <Link to="/voting" className="item">{'Voting'}</Link>
                    </div>
                </div>

            </div>
        );
    }
}

export default Home;
