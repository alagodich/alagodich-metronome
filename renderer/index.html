<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Metronome</title>
    <!-- CSS -->
    <link href="../dist/app.css" rel="stylesheet">
  </head>

  <body>
      <div class="ui main container basic segment">
          <div class="app"></div>
      </div>
  </body>

  <script>
    require('../dist/vendors.js');
    require('../dist/app.js');
  </script>

</html>
