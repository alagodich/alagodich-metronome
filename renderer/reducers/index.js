import {combineReducers} from 'redux';
import {routerReducer as routing} from 'react-router-redux';
import counter from './counter';
import voting from './voting';

const rootReducer = combineReducers({counter, voting, routing});

export default rootReducer;
