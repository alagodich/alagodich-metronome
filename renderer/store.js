import {createStore, applyMiddleware, compose} from 'redux';
import thunk from 'redux-thunk';
import {hashHistory} from 'react-router';
import {routerMiddleware, push} from 'react-router-redux';
import createLogger from 'redux-logger';
import rootReducer from './reducers/index.js';

import * as counterActions from './actions/counter.js';

const actionCreators = Object.assign({
        push
    }, counterActions),
    logger = createLogger({
        level: 'info',
        collapsed: true
    }),
    // If Redux DevTools Extension is installed use it, otherwise use Redux compose
    /* eslint-disable no-underscore-dangle */
    composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Options: http://zalmoxisus.github.io/redux-devtools-extension/API/Arguments.html
        actionCreators
    })
    : compose,
    router = routerMiddleware(hashHistory),
    enhancer = composeEnhancers(
        applyMiddleware(thunk, router, logger)
    );

export default function configureStore(initialState) {
    return createStore(rootReducer, initialState, enhancer);
}
