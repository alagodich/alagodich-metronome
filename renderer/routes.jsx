import React from 'react';
import {Route, IndexRoute} from 'react-router';
import App from './containers/App.jsx';
import HomePage from './containers/HomePage.jsx';
import CounterPage from './containers/CounterPage.jsx';
import {metronomePage, onInit as initMetronome} from './containers/metronomePage.jsx';
import VotingPage from './containers/VotingPage.jsx';

// export default dispatch => (
//     <Route path={'/'} component={App}>
//         <IndexRoute component={HomePage} />
//         <Route path="/counter" component={CounterPage} />
//         <Route path="/metronome" component={metronomePage()} onEnter={initMetronome(dispatch)} />
//         <Route path="/voting" component={VotingPage} />
//     </Route>
// );
export default dispatch => (
    <Route path={'/'} component={App}>
        <IndexRoute component={metronomePage()} onEnter={initMetronome(dispatch)} />
    </Route>
);
