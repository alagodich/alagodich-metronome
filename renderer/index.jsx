/* eslint no-console: 0 */

import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';
import {syncHistoryWithStore} from 'react-router-redux';
import {ipcRenderer} from 'electron';
import {Router, hashHistory} from 'react-router';
import configureStore from './store.js';
import routes from './routes.jsx';

const store = configureStore(),
    history = syncHistoryWithStore(hashHistory, store);

store.dispatch({
    type: 'SET_ENTRIES',
    entries: [
        'Shallow Grave',
        'Trainspotting',
        'A Life Less Ordinary',
        'The Beach',
        '28 Days Later',
        'Millions',
        'Sunshine',
        'Slumdog Millionaire',
        '127 Hours',
        'Trance',
        'Steve Jobs'
    ]
});
store.dispatch({type: 'NEXT'});

ipcRenderer.on('store', (event, arg) => {
    console.log('store-update recieved', event, arg);
});

ipcRenderer.send('window-connected');

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    render(
        <Provider store={store}>
            <Router history={history} routes={routes} />
        </Provider>,
        container
    );
};
