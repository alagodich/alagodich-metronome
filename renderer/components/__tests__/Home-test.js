import React from 'react';
import Home from '../Home.jsx';
import renderer from 'react-test-renderer';

test('Home can be rendered', () => {
    const component = renderer.create(<Home />),
        tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
