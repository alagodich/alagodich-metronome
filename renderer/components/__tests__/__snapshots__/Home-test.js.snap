exports[`test Home can be rendered 1`] = `
<div
  className="ui basic segment">
  <div
    className="ui text container">
    <h1
      className="ui inverted header">
      Home
    </h1>
    <div
      className="ui vertical text menu">
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Counter
      </a>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Metronome
      </a>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Voting
      </a>
    </div>
  </div>
</div>
`;
